lib/rules/no-return-await.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ReturnStatement(node) {
+                if (node.argument && node.argument.type === "AwaitExpression") {
+                    context.report({
+                        node,
```

Comment:
No, `getFirstToken(node)` would get the `return` keyword rather than the `await` keyword.

After trying it out locally, it looks like acorn/espree is incorrectly identifying `await` as an identifier instead of a keyword:

```
Token {
  type: 'Identifier',
  value: 'await',
  start: 32,
  end: 37,
  loc:
   SourceLocation {
     start: Position { line: 3, column: 8 },
     end: Position { line: 3, column: 13 } },
  range: [ 32, 37 ] }
```

So `sourceCode.getFirstToken(node.argument)` is correct, but we should probably remove the type assertion from the tests for the time being.